File: app-service-comments/src/main/java/com/github/arhor/aws/graphql/federation/comments/service/PostService.java
package com.github.arhor.aws.graphql.federation.comments.service;

import com.github.arhor.aws.graphql.federation.comments.generated.graphql.types.Post;

import java.util.UUID;

public interface PostService {

    Post findInternalPostRepresentation(UUID postId);

    void createInternalPostRepresentation(UUID postId, UUID idempotencyKey);

    void deleteInternalPostRepresentation(UUID postId, UUID idempotencyKey);
}

File: dgs-comments/src/main/kotlin/com/github/arhor/dgs/comments/service/PostService.kt
package com.github.arhor.dgs.comments.service

import com.github.arhor.dgs.comments.generated.graphql.types.Post

interface PostService {
    fun getPostById(postId: Long): Post
    fun createPost(postId: Long)
    fun deletePost(postId: Long)
}
